pkg/reconciler/revision/table_test.go

```diff
@@ -471,6 +471,24 @@ func TestReconcile(t *testing.T) {
 			Object: pa("foo", "deploy-timeout", WithReachabilityUnreachable),
 		}},
 		Key: "foo/deploy-timeout",
+	}, {
+		Name: "revision failure because replicaset and deployment failed",
+		// Test just define a Revision so when an error creating the replicaset happens
+		// this is the state that we want to have, everything been but failing due to Revision ReplicaSetFailure
+		Objects: []runtime.Object{
+			Revision("foo", "deploy-replicaset-failure",
+				WithLogURL, MarkActivating("Deploying", ""),
+				WithRoutingState(v1.RoutingStateActive, fc),
+				withDefaultContainerStatuses(),
+				WithRevisionObservedGeneration(1),
+				MarkResourcesUnavailable("FailedCreate", "I ReplicaSet failed!"),
```

Comment:
This currently is just testing the stable state of the failed revision.

I think this test should default to the the revision is in progress - so Ready=Unknown and MarkResourceAvailable to Unknown. Then we assert there is an update that the revision Ready was flipped to False

Reply:
The thing I tried here is that, but since we cannot create the deployment then it bubbles up the failure since the beginning o.o